### PR TITLE
feat: Remove plugin option for logging error events to Sentry

### PR DESCRIPTION
**Pull request by murarustefaan**
Closes [#1809](https://github.com/cloudquery/cloudquery-issues/issues/1809).

### CODE REVIEW

File: serve/plugin.go [hunk 1]
```diff
@@ -306,12 +268,7 @@ func (s *PluginServe) newCmdPluginServe() *cobra.Command {
 	cmd.Flags().StringVar(&otelEndpointURLPath, "otel-endpoint-urlpath", "", "Open Telemetry HTTP collector endpoint URL path")
 	cmd.Flags().StringArrayVar(&otelEndpointHeaders, "otel-endpoint-headers", []string{}, "Open Telemetry HTTP collector endpoint headers")
 	cmd.Flags().BoolVar(&otelEndpointInsecure, "otel-endpoint-insecure", false, "use Open Telemetry HTTP endpoint (for development only)")
-	cmd.Flags().BoolVar(&noSentry, "no-sentry", false, "disable sentry")
```
Comment (murarustefaan) [inline serve/plugin.go:309]: I'm not sure here tbh what would be the best approach, whether to remove this or have it do nothing and mark as deprecated. Wdyt?

Comment (disq) [reply serve/plugin.go:309]: I would keep the option but make it deprecated and no-op.

Comment (murarustefaan) [reply serve/plugin.go:309]: That makes sense, I wasn't sure if deleting it was the best decision as if it was like that, I'd have had to delete it from the other plugin sdks as well.

**Review by candiduslynx — APPROVED** [review]

**Review by erezrokah — APPROVED** [review]

File: serve/plugin.go (hunk 1 above)
Comment (erezrokah) [reply serve/plugin.go:309]: Yeah I think if we remove the flag new plugins might not work with older CLI versions (don't remember if we're doing strict arg parsing)

**Pull request merged**